validate/validators/sam.py

```diff
@@ -0,0 +1,24 @@
+'''Helper methods for SAM file validation'''
+
+from pathlib import Path
+import argparse
+
+import pysam
+
+def validate_sam_file(path):
+    '''Validates sam file'''
+    try:
+        pysam.quickcheck(str(path))
+    except pysam.SamtoolsError as err:
+        raise ValueError("samtools sam check failed. " + str(err)) from err
+
+    sam_head: pysam.IteratorRowHead = pysam.AlignmentFile(str(path)).head(1)
+    if next(sam_head, None) is None: #if the iterator is exhausted, next() returns None
```

Comment:
Consider removing the comment. This behavior is part of the documentation for [`next`](https://docs.python.org/3/library/functions.html#next).